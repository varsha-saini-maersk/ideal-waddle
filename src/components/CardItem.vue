<script setup>
import { defineProps } from 'vue'
const props = defineProps({
  imageUrl: {
    type: String,
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
  height: {
    type: Number,
    required: false,
  },
  weight: {
    type: Number,
    required: false,
  },
})
</script>
<template>
  <div class="card">
    <img :src="imageUrl" alt="Icon" class="card-img" />
    <div class="card-details">
      <h3>{{ `${name} - ${height}cm / ${weight} pounds` }}</h3>
      <p>List of abilities</p>
    </div>
  </div>
</template>

<style scoped>
.card {
  border: 1px solid #ccc;
  border-radius: 8px;
  overflow: hidden;
  width: 250px;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}
.card-img {
  width: 100%;
  height: 150px;
  object-fit: cover;
}

.card-details {
  padding: 10px;
}

@media (min-width: 1024px) {
}
</style>
