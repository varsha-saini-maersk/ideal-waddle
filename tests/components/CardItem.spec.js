import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import CardItem from '@/components/CardItem.vue';

describe('CardItem.vue', () => {
  it('renders correctly with required props', () => {
    const wrapper = mount(CardItem, {
      props: {
        imageUrl: 'test url',
        name: 'Test Name',
        height: 180,
        weight: 75,
      },
    });

    expect(wrapper.find('.card-img').attributes('src')).toBe('test url');
    expect(wrapper.find('.card-header').text()).toBe('Name : Test Name');
  });


  it('matches snapshot', () => {
    const wrapper = mount(CardItem, {
      props: {
        imageUrl: 'https://example.com/image.jpg',
        name: 'Snapshot Test',
        height: 170,
        weight: 65,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
