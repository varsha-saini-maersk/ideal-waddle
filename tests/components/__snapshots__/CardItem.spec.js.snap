// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardItem.vue > matches snapshot 1`] = `
"<div data-v-5877d0fb="" class="card"><img data-v-5877d0fb="" src="https://example.com/image.jpg" alt="Icon" class="card-img">
  <div data-v-5877d0fb="" class="card-details">
    <h3 data-v-5877d0fb="">Snapshot Test - 170cm / 65 pounds</h3>
    <p data-v-5877d0fb="">List of abilities</p>
  </div>
</div>"
`;
